src/v/kafka/server/connection_context.h

```diff
@@ -416,7 +437,9 @@ class connection_context final
      * A map keeping virtual connection states, during default operation the map
      * is empty
      */
-    absl::node_hash_map<bytes, ss::lw_shared_ptr<virtual_connection_state>>
+    absl::node_hash_map<
+      virtual_connection_id,
+      ss::lw_shared_ptr<virtual_connection_state>>
```

Comment:
I seem to recall that the slot array on `node_hash_map` can grow quite large if we have lots of buckets. Could that be a concern here if there are many more virtual connection IDs compared to the usual number of live connections?

Reply:
i will change that to chunked hash map